metabase/metabase
# Add opt-in CC option to Admin Email Setting

**Problem:** Some users have firewalls setup that prevent sending emails as BCC. This prevents user from being able to use our email subscriptions feature. This PR adds a front-end option in admin settings > email to opt-in to sending emails as CC

**Context:**
- https://github.com/metabase/metabase/issues/34293

---
**Desired Behavior**/**Acceptance Criteria**

_Behaviors_
- [ ] Admin email settings now has a CC opt-in setting
  ![image](https://github.com/metabase/metabase/assets/22608765/21dcf931-f59f-46f6-92be-bf5c888c9187)


_Tests_
- [x] With BCC option enabled (default), email subscriptions are sent as BCC
- [x] With CC option enabled, email subscriptions are sent as CC

---
**How to Verify**

1. Log in as admin
2. Setup SMTP (see [maildev](https://github.com/maildev/maildev) for first time setup)
3. Go to a dashboard > subscriptions > email it
![Screenshot from 2023-11-07 22-49-24](https://github.com/metabase/metabase/assets/22608765/a238c0ad-e9a9-41a6-a371-55afd73fd52b)
4. Add a user in the `to` field
5. Click send email now
6. Open up mailbox
7. Click on email
8. See that it is sent as BCC
![Screenshot from 2023-11-07 22-37-43](https://github.com/metabase/metabase/assets/22608765/6d700b57-8a1a-4299-983f-e792760bbdd2)
9. Go to admin settings > email
10. Toggle CC opt-in
![image](https://github.com/metabase/metabase/assets/22608765/8463e95d-94ad-4133-af11-7c8e5a2782d3)
11. Go back to dashboard subscription
12. Setup and send email again
13. Open up mailbox and click on email
14. See that it is sent as CC (no BCC field)
![Screenshot from 2023-11-07 22-37-47](https://github.com/metabase/metabase/assets/22608765/7f02bc82-308e-4b5d-9622-3036e4292175)
